mm2src/mm2_core/src/sql_connection_pool.rs

```diff
@@ -0,0 +1,332 @@
+use crate::mm_ctx::{log_sqlite_file_open_attempt, path_to_dbdir, MmCtx};
+use async_std::sync::RwLock as AsyncRwLock;
+use common::log::error;
+use db_common::async_sql_conn::AsyncConnection;
+use db_common::sqlite::rusqlite::Connection;
+use futures::channel::mpsc::{channel, Receiver, Sender};
+use futures::lock::Mutex as AsyncMutex;
+use gstuff::try_s;
+use std::collections::HashMap;
+use std::path::PathBuf;
+use std::sync::{Arc, Mutex, MutexGuard, RwLock};
+
+pub const ASYNC_SQLITE_DB_ID: &str = "KOMODEFI.db";
+const SYNC_SQLITE_DB_ID: &str = "MM2.db";
+const SQLITE_SHARED_DB_ID: &str = "MM2-shared.db";
+
+/// Represents the kind of database connection ID: either shared or single-user.
```

Comment:
Should this description not a bit better clarify what shared and single-user is?
I guess 'shared' means that this db is shared between accounts and 'single' is used for a single account, right?

Reply:
BTW, if we pass db_id == None to sqlite_conn_opt fn (which should return a single-user connection), would this mean that we in fact will use db as shared?

Reply:
Nop, it for backward compatibility and startup `db_id`. passing `None` basically means we are using the startup `db_id`

Reply:
> Should this description not a bit better clarify what shared and single-user is?

updated